chains/astar/server/src/lib.rs

```diff
@@ -26,7 +26,14 @@ impl BlockchainClient for AstarClient {
     }
 
     async fn new(config: BlockchainConfig, addr: &str) -> Result<Self> {
-        let client = EthereumClient::new(config, addr).await?;
+        // TODO: Fix this hack, need to support multiple addresses per node
+        let mut ethereum_rpc_uri = NodeUri::parse(addr)?;
+        ethereum_rpc_uri.scheme = if ethereum_rpc_uri.scheme == "ws" {
+            "http"
+        } else {
+            "https"
+        };
+        let client = EthereumClient::new(config, ethereum_rpc_uri.to_string().as_str()).await?;
```

Comment:
Astar requires both `ws` and `http` connections, using this dirty solution for now.